articles/npm-packages/index.md

```diff
@@ -0,0 +1,191 @@
+
+```json
+{
+  "name": "clist-node",
+  "version": "1.0.0"
+}
+```
+
+- In our sample NPM package, we are going to use the axios package for making API calls. Install it by using this command
+
+```bash
+npm i axios
+```
+
+- We'll be using the [clist API](https://clist.by/) that returns a list of contests like hackathons, programming contests, etc..,
+
+- Create an account on the [clist website](https://clist.by/api/v1/doc/) and get an API key.
+
+- The entry point of our package would be the **`index.js`** file. Create a file named index.js in the same directory.
+
+```js
+// module.exports exports the function getContests as a promise and exposes it as a module.
+// we can import an exported module by using require().
+module.exports = async function getContests() {
+  const axios = require("axios"); // Importing the Axios module to make API requests
+  var result;
+
+  const username; // Insert Your Username here
+  const api_key; //Insert API key here
+
+  await axios // Making a GET request using axios and requesting information from the API
+    .get(
+      "https://clist.by/api/v1/json/contest/?username=" + username + "&api_key=" + api_key + "&limit=20&end__gt=2020-09-19T00%3A00%3A00"
+    )
+    .then((response) => { // If the GET request is successful, this block is executed
+      return response; // The response of the API call is passed on to the next then block
+    })
+    .then((contests) => { // In this block, we store the response data into a variable 'result'
+      result = contests.data.objects;
+    })
+    .catch((err) => {
+      console.log(err); // Error handler
+    });
+  return result; // The contest data is returned
+};
+```
+
+Now that we've programmed the package, it's time to deploy it.
+
+### Deployment
+
+- To deploy the package, enter the command,
+
+```bash
+npm publish
+```
+
+![NPM Publish](/engineering-education/npm-packages/npm-publish.png)
+
+- In a while, you'll get an email from the NPM registry indicating that your package has been successfully published!
+
+### Using the Package
+
+- Let us install and test our package.
+
+- Open a new folder and initialize a **`package.json`** by using the command,
+
+```bash
+npm init -y
+```
+
+_Note: The -y option creates a package.json without an interactive process_
+
+- Now, let us install our published npm package.
+
+```bash
+npm i clist-node
+```
+
+- Create an index.js file and let's code.
+
+```js
+const clist = require("clist-node"); // import the installed package
+
+clist()
+  .then((res) => {
+    // If the function successfully retrieves the data, it enters this block
+    console.log(res); // Print the contest data on the console
+  })
+  .catch((err) => {
+    console.log(err); // Error handler
+  });
+```
+
+- Run this file by using the command,
+
+```bash
+node index.js
+```
+
+- The contest data from our NPM package would be printed out on the console. We can do various other things using the retrieved data.
```

Comment:
Perhaps provide a few suggestions on what readers could do with the retrieved data